packages/web3-eth-abi/src/api/parameters_api.ts

```diff
@@ -172,14 +114,11 @@ export const decodeParametersWith = (
 					'or querying a node which is not fully synced.',
 			);
 		}
-		const res = ethersAbiCoder.decode(
-			mapTypes(abis).map(p => ParamType.from(p)),
-			`0x${bytes.replace(/0x/i, '')}`,
-			loose,
-		);
-		return formatArrayResToObject(res);
+		return decodeParametersInternal(abis, `0x${bytes.replace(/0x/i, '')}`, loose);
 	} catch (err) {
-		throw new AbiError(`Parameter decoding error: ${(err as Error).message}`);
+		throw new AbiError(`Parameter decoding error: ${(err as Error).message}`, {
+			internalErr: err,
+		});
```

Comment:
Could we utilize here `innerError` at the base class `BaseWeb3Error`?
```suggestion
		throw new AbiError(`Parameter decoding error: ${(err as Error).message}`, {}, err);
```